Identity/Identity.BusinessLogic/Services/Implementations/TokenService.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Identity.BusinessLogic.Exceptions;
+using Identity.BusinessLogic.Models;
+using Identity.BusinessLogic.Models.TokenService;
+using Identity.BusinessLogic.Options;
+using Identity.BusinessLogic.Services.Interfaces;
+using Identity.DataAccess.Entities;
+using Identity.DataAccess.Repositories.Interfaces;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Identity.BusinessLogic.Services.Implementations
+{
+    public class TokenService : ITokenService
+    {
+        private readonly JwtOptions _jwtOptions;
+        private readonly ITokenRepository _tokenRepository;
+
+        public TokenService(
+            IOptions<JwtOptions> jwtOptions,
+            ITokenRepository tokenRepository)
+        {
+            _jwtOptions = jwtOptions.Value;
+            _tokenRepository = tokenRepository;
+        }
+
+        public async Task<Tokens> GetTokensAsync(GetTokensRequest request)
+        {
+            var identity = GetClaimsIdentity(request);
+            var tokens = new Tokens
+            {
+                AccessToken = GenerateAccessToken(identity),
+                RefreshToken = await CreateRefreshTokenAsync(request.Id)
+            };
+
+            return tokens;
+        }
+
+        public async Task<Tokens> UseRefreshTokenAsync(Tokens tokens)
+        {
+            await ValidateRefreshToken(tokens.RefreshToken);
+            var identity = await GetIdentityFromTokenAsync(tokens.AccessToken);
+            var userId = identity.Claims
+                .Where(claim => claim.Type == ClaimTypes.NameIdentifier)
+                .Select(claim => claim.Value)
+                .FirstOrDefault()!;
+
+            var newTokens = new Tokens
+            {
+                AccessToken = GenerateAccessToken(identity),
+                RefreshToken = await CreateRefreshTokenAsync(userId)
+            };
+
+            return newTokens;
+        }
+
+        public async Task<ClaimsIdentity> GetIdentityFromTokenAsync(string token)
+        {
+            var handler = new JwtSecurityTokenHandler();
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtOptions.Key));
+            var parameters = new TokenValidationParameters
+            {
+                ValidateIssuer = false,
```

Comment:
you can move your validation parameters to AddAuthenication() method which is located in program.cs class or in extension method

Reply:
With AddAuthenication() I specified the default jwt authentication scheme with it's own parameters, but here I need to validate the signing key only to extract user claims, should I move these parameters to program.cs and specify another scheme or maybe just create a private method GetValidationParameters in the TokenService?